tests/test_geometries.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+"""
+Purpose: Unit testing for geometries.py
+Author : Chris Schnaufer <schnaufer@arizona.edu>
+Notes:
+    This file assumes it's in a subfolder off the main folder
+"""
+import json
+import os
+
+from osgeo import ogr, osr
+from shapely.geometry import Point, LineString, MultiPoint
+
+from agpypeline import geometries
+
+TEST_FILES = ['agpypeline/geometries.py']
+JSON_FILE = os.path.realpath('data/geometries.json')
+
+
+def test_exists():
+    """Tests whether all necessary files are accessible"""
+    for file in TEST_FILES:
+        assert os.path.isfile(file)
+    assert os.path.isfile(JSON_FILE)
+
+def test_geometries_calculate_centroid_from_wkt():
+    """Tests calculate_centroid_from_wkt on a few basic geometries from shapely.geometry"""
+    test_point = Point(0, 0)
+    wkt = test_point.centroid.wkt
+    result = geometries.calculate_centroid_from_wkt(wkt)
+    assert result == (0.0, 0.0)
+    line = LineString([(2, 0), (2, 4), (3, 4)])
+    wkt = line.wkt
+    result = geometries.calculate_centroid_from_wkt(wkt)
+    assert result == (2.1, 2.4)
+    multi = MultiPoint([(0, 0), (1, 1), (1, -1), (1, 0)])
+    wkt = multi.wkt
+    result = geometries.calculate_centroid_from_wkt(wkt)
+    assert result == (0.75, 0.0)
+
+
+def test_geometries_calculate_overlap_percent():
+    """Tests calculate_overlap_percent by trying to overlap the bounds from a json file with themselves and then from
+    a modified version of themselves"""
+    loaded_file = json.load(open(JSON_FILE, 'r'))
+    check_bounds = loaded_file["features"][0]["geometry"]
+    bounding_box = {"type": "Polygon", "coordinates": [
+        [[408989, 3659975], [408990, 3659975], [408990, 3659972], [408989, 3659972], [408989, 3659975]]]}
+    assert geometries.calculate_overlap_percent(check_bounds, check_bounds) == 1.0
+    assert round(geometries.calculate_overlap_percent(check_bounds, bounding_box), 4) == 0.7591
+
+
+def test_geometries_convert_geometry():
+    """Tests convert_geometry by checking the function call on geometries contained within a loaded .json file
+    against a file containing function call results"""
+    null_input_test = geometries.convert_geometry(None, None)
+    assert null_input_test is None
+    loaded_file = json.load(open(JSON_FILE, 'r'))
+    check_result = json.load(open("data/convert_geometry.json", 'r'))
+    for feature in range(len(loaded_file["features"])):
```

Comment:
The only thing I can think of adding here would be to call the conversion function with the same EPSG code as an original polygon. This would cover the case where the called function optimizes its actions by comparing the codes and returning the original if they're the same. I think only having one call with this setup and checking the results would work. If you're ready to move on, please create an issue for this